### ISSUE TITLE
About the CIDEr  Score Problem

### ISSUE THREAD

**Issue by mymuli** [LuoweiZhou/e2e-gLSTM-sc#10]
When I finished training e2e-gLSTM and e2e-gLSTM-SC, the CIDEr score was only about 0.56. ..How can I fine-tune it to reach above 0.9? Can you give me some advice? Thank you very much.

**Comment by mymuli** [comment]
I use VGG-16 to train.

According to the paper, for the first two steps, my training commands are as follows：
（1）th train_new.lua  -language_eval 1  -max_iters 100010  -cnn_learning_rate 0.00001 -cnn_weight_decay 0.001
（2）th train_new.lua  -language_eval 1  -max_iters 150010 -start_from ckpt-train-val-fine-tuning/model_id.t7 -finetune_cnn_after 0 -cnn_learning_rate 0.00001 -cnn_weight_decay 0.001 -learning_rate_decay_every 25000 -learning_rate_decay_start 1

Good results have been obtained，Bleu_1:0.664, CIDEr:0.759


In your paper：“3) train td-gLSTM with initializd text-conditional matrix but without CNN fine-tuning for 150,000 iterations.” 

From the results of the first step, I went on to train, but the effect was very poor，CIDEr<0.759. The training order was as follows：
th train_sc.lua -start_from ckpt-train-val/model_id.t7 -language_eval 1 -language_model 'misc_tc.LanguageModel_sc' -max_iters 150010

I started training from scratch and the results were poor, CIDEr<0.759. The training orders are as follows：
th train_sc.lua  -language_model 'misc_tc.LanguageModel_sc' -language_eval 1  -max_iters 150010  -cnn_learning_rate 0.00001 -cnn_weight_decay 0.001

Can you tell me how to train Step 3?

**Comment by LuoweiZhou** [comment]
Hi @Rmuli, sorry for the delay. 75.9 is still low even for the end-to-end gLSTM. Please refer to the [neuraltalk2](https://github.com/karpathy/neuraltalk2) repo to see if you can get 0.9-1.0 CIDEr first. A few tips might be helpful for your debugging: i) make sure your image input values are valid (e.g., whether between 0-255 or 0-1, normalization), ii) try using ResNet since it generally gives better performance.
Besides, this Torch repo is no longer under maintenance. To get timely support, here are some resources you might find helpful:
[our recent GVD work](https://github.com/facebookresearch/grounded-video-description)
[neural baby talk](https://github.com/jiasenlu/NeuralBabyTalk)
[adaptive attention](https://github.com/jiasenlu/AdaptiveAttention)

**Comment by mymuli** [comment]
Hi @LuoweiZhou , In accordance with your suggestion, I have carried out the experiment and achieved good results. ...Thank you very much!

**Comment by LuoweiZhou** [comment]
Great to know that. Thanks for your interest in our work.
